docs/build-instructions.md

```diff
@@ -2,126 +2,202 @@
 ```
 
+* Change PCCS server IP
+``` shell
+vim /etc/sgx_default_qcnl.conf
+```
+``` vi
+# PCCS server address
+PCCS_URL=https://1.2.3.4:8081/sgx/certification/v3/ (your pccs IP)
+
+# To accept insecure HTTPS certificate, set this option to FALSE
+USE_SECURE_CERT=FALSE
+```
+
+### Build eHSM
+
 * Build the eHSM-KMS
 ```shell
-$ git clone https://github.com/intel/ehsm.git ehsm && cd ehsm
+git clone https://github.com/intel/ehsm.git ehsm && cd ehsm
 
-$ make
+make
 
-Notes: you can find the services binaries under the bin/ folder 
+Notes: you can find the services binaries under the out/ehsm-core folder 
 ```
 * Unittest w/o REST Call
 ```shell
-$ cd bin/ehsm-core
-$ ./ehsm_core_test
+cd out/ehsm-core
+sudo ./ehsm_core_test
 ```
 Then, you will get the below test result:<br>
 ![unittest-result-without-rest.png](diagrams/unittest-result-without-rest.PNG)
 
 
-* Unittest w/ REST Call
+* Enable REST APIs
 ```shell
-# Install nodejs-16
-$  sudo apt install -y curl
-$ curl -fsSL https://deb.nodesource.com/setup_16.x | sudo -E bash -
-$ sudo apt update
-$ sudo apt install -y nodejs
-$ node --version
+#install nodejs-16
+sudo apt install -y curl
+curl -fsSL https://deb.nodesource.com/setup_16.x | sudo -E bash -
+sudo apt update
+sudo apt install -y nodejs
+
+node --version
 v16.1.0
 
-# Copy the required *.so to the ehsm_kms_service
-$ cd ehsm_kms_service
-$ cp ../bin/ehsm-core/libehsmnapi.so .
-$ cp ../bin/ehsm-core/libenclave-ehsm-core.signed.so .
+#copy the required *.so to the ehsm_kms_service
+cd ehsm_kms_service
+cp ../out/ehsm-core/libehsmnapi.so .
+cp ../out/ehsm-core/libenclave-ehsm-core.signed.so .
+
+#install the required npm js libs for the kms service
+npm install
+```
+
+* Install CouchDB
+```shell
+docker pull couchdb
+
+#change the user, password, and port here if needed
+docker run --name my-couchdb -e COUCHDB_USER=admin -e COUCHDB_PASSWORD=password -p 5984:5984 -d couchdb
+```
+
+* Export your user, passwork, IP, and port to root environment (these params need change)
+``` shell
+sudo su (if not root now)
+export EHSM_CONFIG_COUCHDB_USERNAME="admin"
+export EHSM_CONFIG_COUCHDB_PASSWORD="password"
+export EHSM_CONFIG_COUCHDB_SERVER="1.2.3.4"
+export EHSM_CONFIG_COUCHDB_PORT="5984"
+```
+``` shell
+#start the kms service (if you got "Error: listen EADDRINUSE: address already in use :::9000", please close the previous service or containers)
+sudo node ehsm_kms_server.js &
+```
 
-# Install the required npm js libs for the kms service
-$ npm install
+## Run the unittest cases (you can do it in another remote device)
+* Need [Install requirement tools](#install-requirement-tools)
+* Enroll your appid and apikey
+``` shell
+cd ehsm_kms_service/enroll_app
+make
+
+cd ../../out/ehsm-kms_enroll_app
+./ehsm-kms_enroll_app http://1.2.3.4:9000/ehsm/ (change to your ip and port)
+```
+You will get below results:<br>
+![image](https://user-images.githubusercontent.com/59001339/159232174-f5c8677c-f3a6-43cd-a1b9-dbe0dd7b8bbd.png)
```

Comment:
if you links to a picture, please upload it to [dock](https://github.com/intel/ehsm/tree/main/docs/diagrams)